docs/version.md

```diff
@@ -0,0 +1,81 @@
+# Semantic Versioning
+
+At the core of `changelog-gen` is the version detection and generation logic,
+check [here](https://semver.org) for details on semantic versioning, and
+[here](https://datatracker.ietf.org/doc/html/rfc2119) for a more detailed look
+at the underlying RFC.
+
+## Version generation
+
+`changelog-gen` uses conventional commit types and its own configuration to
+determine what type of release to trigger based on the commits since the last release.
+
+### Major
+
+Any breaking change will trigger a major release, unless the library is in then
+initial `0.y.z` development release. In commit terms, any commit message with a
```

Comment:
This sounds weird .. maybe:
> unless the package is in the development release phase, i.e. the current version follows the pattern `0.y.z`.

package instead of library to make it more generic ?